Gordon360/Controllers/AccountsController.cs

```diff
@@ -123,26 +123,23 @@ public async Task<ActionResult<IEnumerable<BasicInfoViewModel>>> SearchWithSpace
             string? department,
             string? building)
         {
-            var viewerGroups = AuthUtils.GetGroups(User);
-
-            // Only students and FacStaff can search students
-            if (accountTypes.Contains("student") && !(viewerGroups.Contains(AuthGroup.Student) || viewerGroups.Contains(AuthGroup.FacStaff)))
-            {
-                accountTypes.Remove("student");
-            }
-
-            var searchResults = _accountService.AdvancedSearch(accountTypes,
-                                                               firstname?.ToLower() ?? "",
-                                                               lastname?.ToLower() ?? "",
-                                                               major ?? "",
-                                                               minor ?? "",
-                                                               hall ?? "",
-                                                               classType ?? "",
-                                                               homeCity?.ToLower() ?? "",
-                                                               state ?? "",
-                                                               country ?? "",
-                                                               department ?? "",
-                                                               building ?? "");
+            IEnumerable<AuthGroup> viewerGroups = AuthUtils.GetGroups(User);
+
+            var accounts = _accountService.GetAccountsToSearch(accountTypes, viewerGroups, homeCity ?? "");
+
+            var searchResults = _accountService.AdvancedSearch(
+                accounts,
+                firstname?.ToLower() ?? "",
+                lastname?.ToLower() ?? "",
+                major ?? "",
+                minor ?? "",
+                hall ?? "",
+                classType ?? "",
+                homeCity?.ToLower() ?? "",
+                state ?? "",
+                country ?? "",
+                department ?? "",
+                building ?? "");

```

Comment:
Shouldn't these all be .ToLower() ? I know our UI validates the ones that are currently not so with select boxes, but we shouldn't be modeling the API with the UI in mind, right?

Reply:
I think the better solution would be to use a case-insensitive string comparison, now that I'm thinking about it. I only changed the formatting here, so I didn't actually consider how best to handle this. 

In theory, the client is selecting from the list of available options (in a select on our frontend in this case), and searching for an invalid option is expected to return no results. But perhaps it is better to just default to case-insensitive comparison for all search fields.